### PR TITLE
feat(integrations): add support for OpenAi

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Add support for OpenAi

## Issue ticket number and link
[EXT-158](https://linear.app/nango/issue/EXT-158/add-support-for-openapi-api-key)

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -2214,6 +2214,22 @@ one-note:
...
     alias: microsoft
     docs: https://docs.nango.dev/integrations/all/one-note
+openai:
+    categories:
+        - productivity
+        - dev-tools
+    auth_mode: API_KEY
+    proxy:
+        base_url: https://api.openai.com
+        headers:
```
Comment (hassan254-prog) [inline packages/shared/providers.yaml:2238]: For users who belong to multiple organizations or access their projects through a legacy user API key, a header can be passed to specify which organization and project should be used for an API request. This is optional; however, introducing `connectionConfig` in Nango would make it a required field. Perhaps we can make `connectionConfig` optional once defined in the `providers.yaml`.
```
curl https://api.openai.com/v1/models \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -H "OpenAI-Organization: org-IAA3Q093FBg7duWPcrysKyly" \
  -H "OpenAI-Project: $PROJECT_ID"
  ```

**Review by TBonnin — APPROVED** [review]

File: packages/shared/providers.yaml (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/providers.yaml:2238]: Can you add information on this to the documentation under gotchas?

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Docs update requested

File: docs-v2/integrations/dev-tools.mdx [hunk 2]
```diff
@@ -23,6 +23,7 @@ sidebarTitle: Developer Tools
     <Card title="Google Play" href="/integrations/all/google-play" color="#68a063" />
     <Card title="Klipfolio" href="/integrations/all/klipfolio" color="#68a063" />
     <Card title="Mailgun" href="/integrations/all/mailgun" color="#68a063" />
+    <Card title="OpenAi" href="/integrations/all/openai" color="#68a063" />
```
Comment (bodinsamuel) [inline docs-v2/integrations/dev-tools.mdx]:
```suggestion
    <Card title="OpenAI" href="/integrations/all/openai" color="#68a063" />
```

File: docs-v2/integrations/all/openai.mdx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+title: OpenAi
+sidebarTitle: OpenAi
```
Comment (bodinsamuel) [inline docs-v2/integrations/all/openai.mdx]:
```suggestion
title: OpenAI
sidebarTitle: OpenAI
```

Comment (bodinsamuel) [reply docs-v2/integrations/all/openai.mdx]: (not going to comment all cases, but it's applicable to many places)

**Review by bodinsamuel — APPROVED** [review]

File: docs-v2/integrations/dev-tools.mdx (hunk 2 above)
Comment (hassan254-prog) [reply docs-v2/integrations/dev-tools.mdx]: I have updated this with the latest [commit](https://github.com/NangoHQ/nango/pull/2755/commits/9a79023abff923496f6d067e7d18e35814a51c46)

File: packages/shared/providers.yaml (hunk 1 above)
Comment (hassan254-prog) [reply packages/shared/providers.yaml:2238]: I have updated this with the latest [commit](https://github.com/NangoHQ/nango/pull/2755/commits/9a79023abff923496f6d067e7d18e35814a51c46)

**Review by bodinsamuel — APPROVED** [review]

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**